base/applications/clipbrd/clipbrd.c

```diff
@@ -389,20 +448,21 @@ static LRESULT WINAPI MainWndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lP
     {
         case WM_CREATE:
         {
-            TEXTMETRICW tm;
-            HDC hDC = GetDC(hWnd);
-
-            /*
-             * Note that the method with GetObjectW just returns
-             * the original parameters with which the font was created.
-             */
-            if (GetTextMetricsW(hDC, &tm))
-            {
-                Globals.CharWidth  = tm.tmMaxCharWidth; // tm.tmAveCharWidth;
-                Globals.CharHeight = tm.tmHeight + tm.tmExternalLeading;
-            }
-            ReleaseDC(hWnd, hDC);
-
+            /* Create a text box */
+            const DWORD style = WS_CHILD | ES_MULTILINE | ES_READONLY | WS_HSCROLL | WS_VSCROLL;
+            Globals.hwndText = CreateWindowExW(WS_EX_CLIENTEDGE,
```

Comment:
I'm not a fan of this at all.

Reply:
Can you give me the reason?

Reply:
The pros of using a textbox (of course, only shown when it's text being displayed, otherwise it must be hidden) is that, indeed, we would get out-of-the-box correct formatting for the text (including for right-to-left languages, ligatures and stuff for e.g. arabic, etc.)

What I really wonder are the cons. Namely, that:
- it allows selecting the text for copying (completely useless and would interfere with clipboard contents), unless of course it is possible to disable that somehow;
- that it _might_ require to duplicate the whole text being displayed in order to have correct scrolling information.

As I've already mentioned in the previous PR, I don't want our clipboard to take a ton of memory whenever a (very) large chunk of text has been copied in the clipboard and is being shown.

Reply:
EM_SETHANDLE would almost work except I think it wants LocalAlloc instead of GlobalAlloc that you are probably using with the clipboard.

Reply:
In Win10, the EDIT control of ANSI version doesn't support ANSI `EM_SETHANDLE`.